src/names/common.h

```diff
@@ -485,4 +485,16 @@ class CNameCache
 
 };
 
+struct NameNewReturn
+{
+    bool ok;
+    std::string err_msg;
+    std::string toaddress;
+    std::string hex;
+    std::string rand;
+    std::string data;
```

Comment:
I think these fields should not be strings.  Rather, `toaddress` could be a `CScript`, `hex` could be a `uint256` and others (`rand` at least) could be `valtype`.